Add nuvo media_player component

ejonesnospam opened the pull request:
Hi! - [FYI, First submission to HA!] Adds new component to home assistant for Nuvo audio amplifier (similar to monoprice component) via serial data connection. Has been tested on HA 0.86.4 for approx 3 months. Dependency (pynuvo) has been published to PyPi.

## Breaking Change:

N/A - Not a breaking change (I hope!)

## Description:
Adds new component to home assistant for Nuvo audio amplifier (similar to monoprice component) via serial data connection.

**Related issue (if applicable):** fixes #<home-assistant issue number goes here>

**Pull request in [home-assistant.io](https://github.com/home-assistant/home-assistant.io) with documentation (if applicable):** home-assistant/home-assistant.io#<home-assistant.io PR 22694>

## Example entry for `configuration.yaml` (if applicable):
```
media_player:
  - platform: nuvo
    port: /dev/ttyUSB0
    model: concerto
    zones:
      1:
        name: Main Bedroom
      2:
        name: Living Room
      3:
        name: Kitchen
      4:
        name: Bathroom
      5:
        name: Dining Room
      6:
        name: Guest Bedroom
    sources:
      1:
        name: Sonos
      2:
        name: Chromecast

```

## Checklist:
  - [ ] The code change is tested and works locally.
  - [ ] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [ ] There is no commented out code in this PR.

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code communicates with devices, web services, or third-party tools:
  - [ ] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [ ] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [ ] New or updated dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [ ] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard/__init__.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard/__init__.py#L23

Commit:
temp to create folder

Commit:
Adds nuvo component

Commit:
Delete junk

homeassistant commented:
Hi @ejonesnospam, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#22694).

Once you do that we will be able to review and accept this pull request.

Thanks!

Commit:
Adds nuvo component

homeassistant commented:
Hi @ejonesnospam, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#22694).

Once you do that we will be able to review and accept this pull request.

Thanks!

Commit:
Adds nuvo component

homeassistant commented:
Hi @ejonesnospam, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#22694).

Once you do that we will be able to review and accept this pull request.

Thanks!

ejonesnospam commented:
Still learning, sorry for my lack of knowledge. Thanks for your patience!!

Commit:
1 blank line to 2 blank lines ;)

Commit:
removal of whitespace

Commit:
added whitespace :\

Commit:
Update media_player.py

Alterations from HA 0.86.4 to 0.90. Locations of some imported constants changed.

Commit:
C0325: Unnecessary parens

ejonesnospam commented:
All checks passed, let me know if I need to do anything else

awarecan commented:
It is harsh time for new integration as we are doing refactoring, we are now requesting all components has to have a [manifest file](https://github.com/home-assistant/architecture/issues/179).

Please rebase you branch against current `dev` branch, and create a `manifest.json` file under your component's folder.

You may need to do more rebase in the next few days as we are still in the progress of refactoring.

ejonesnospam commented:
Well, any changes to 'components/monoprice/media_player.py' will be mirrored into this new component. Also, I pulled from the latest home-assistant dev branch yesterday, so I'm not clear why I would need to rebase again so quickly.

awarecan commented:
We just added `mainifest.json` to every each component. And required all new component follow it. We are moving fast 🚀 

The pull request template and dev doc will adjust shortly.

ejonesnospam commented:
Well, since it hasn't happened yet, I followed the guidelines that were posted at the time. Feel free to do with this as you wish. Clearly, your work and effort is much more important than mine.

awarecan commented:
Sorry but It already happened. Your code would not be able to merge unless you rebased. Otherwise dev branch will broke.

Of course, you don't have to rush to do this.

ejonesnospam commented:
And you or someone else can do with it as you wish. It certainly won't be me. Your attitude clearly says I should invest my energy elsewhere

robbiet480 commented:
Hi there @ejonesnospam, 

Thanks for your contribution to Home Assistant! Sadly, as @awarecan stated, we do have new requirements for submitting integrations. Yes, they are not yet documented, but will be very soon. I’ll also apologize here that you are being caught up in this now, it’s just bad timing you couldn’t have avoided. We could have also done better to communicate this change by updating the pull request template last night after we added manifests to all 800+ or whatever integrations. 

If you want, you can wait until there is documentation on the manifest in the next few days, or you can get ahead of it and just take care of it now and see your PR merged in the next few hours. It’s your decision how to proceed. But this PR will not be merged without a manifest file. It’s really easy to do, I swear. It’s just a JSON file with maybe 10 lines total. @awarecan even linked a helpful example for you above. 

Now, what we won’t stand for is the tone that you took with @awarecan. He was totally cordial to you and, from what I see, did nothing to upset you other than politely inform you of a new requirement that will take maybe 3 minutes to add. I would respectfully request you apologize to @awarecan so that we can all just move on and see this excellent contribution get merged into Home Assistant, making you just the latest developer to join the small club of Home Assistant contributors. I will also remind you that by submitting this PR and signing the CLA you implicitly agreed to follow our [Code of Conduct](https://www.home-assistant.io/code_of_conduct/), which it seems you may have violated by your conduct in this PR. Alternatively, you can continue with your same tone and see this PR closed permanently and yourself potentially banned from our GitHub accounts. Again, your choice!

Thanks again for your contribution. Please let me know if you have any questions.

Thr33D commented:
@ejonesnospam I can only ask as nicely and professionally as I can...that you continue to develop this component.  There are quite a few of us out here with the Nuvo product that would absolutely welcome this integration.  While I certainly do not possess the programming knowledge and skills, it's obvious that you do.  This integration (which I understand was you first) was probably the most exciting I have seen so far to save my aging Nuvo MPS4.  And from reading the thread, the requirements are minor compared to the work you have already dedicated to this.

I sincerely hope you reconsider as you certainly have the skills to be a significant contributor to this home assistant endeavor.

MartinHjelmare commented:
I'll close this PR since it has gone stale. Please open a new PR when you have time to finish.
